docs/advanced_topics/boundblocks_and_values.rst

```diff
@@ -0,0 +1,113 @@
+.. _boundblocks_and_values:
+
+About BoundBlocks and values
+----------------------------
+
+All StreamField block types accept a ``template`` parameter to determine how they will be rendered on a page. However, for blocks that handle basic Python data types, such as ``CharBlock`` and ``IntegerBlock``, there are some limitations on where the template will take effect, since those built-in types (``str``, ``int`` and so on) cannot be 'taught' about their template rendering. As an example of this, consider the following block definition:
+
+.. code-block:: python
+
+    class HeadingBlock(blocks.CharBlock):
+        class Meta:
+            template = 'blocks/heading.html'
+
+where ``blocks/heading.html`` consists of:
+
+.. code-block:: html+django
+
+    <h1>{{ value }}</h1>
+
+This gives us a block that behaves as an ordinary text field, but wraps its output in ``<h1>`` tags whenever it is rendered:
+
+.. code-block:: python
+
+    class BlogPage(Page):
+        body = StreamField([
+            # ...
+            ('heading', HeadingBlock()),
+            # ...
+        ])
+
+.. code-block:: html+django
+
+    {% load wagtailcore_tags %}
+
+    {% for block in page.body %}
+        {% if block.block_type == 'heading' %}
+            {% include_block block %}  {# This block will output its own <h1>...</h1> tags. #}
+        {% endif %}
+    {% endfor %}
+
+This kind of arrangement - a value that supposedly represents a plain text string, but has its own custom HTML representation when output on a template - would normally be a very messy thing to achieve in Python, but it works here because the items you get when iterating over a StreamField are not actually the 'native' values of the blocks. Instead, each item is returned as an instance of ``BoundBlock`` - an object that represents the pairing of a value and its block definition. By keeping track of the block definition, a ``BoundBlock`` always knows which template to render. To get to the underlying value - in this case, the text content of the heading - you would need to access ``block.value``. Indeed, if you were to output ``{% include_block block.value %}`` on the page, you would find that it renders as plain text, without the ``<h1>`` tags.
+
+(More precisely, the items returned when iterating over a StreamField are instances of a class ``StreamChild``, which provides the ``block_type`` property as well as ``value``.)
```

Comment:
Think we should add backticks around ``StreamField`` in this context as it's referring to the field type in code.